utilities/infra.py

```diff
@@ -583,10 +584,31 @@ def verify_no_failed_pods(
         isvc (InferenceService): InferenceService object
         runtime_name (str): ServingRuntime name
         timeout (int): Time to wait for the pod.
+        timeout_wait_for_pods (int): Time to wait for the pods.
+
     Raises:
-            FailedPodsError: If any pod is in failed state
+        FailedPodsError: If any pod is in failed state
+        TimeoutExpiredError: If no pods were created within `timeout_wait_for_pods` seconds.
 
     """
+    LOGGER.info("Wait for pods creation")
+
+    try:
```

Comment:
May be this should be part of another function and this function should be called only when the previous one passes? I don't really like the idea of a function doing many things. Triage becomes messy as it grows.

Reply:
done